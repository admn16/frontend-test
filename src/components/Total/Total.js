import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Card from 'components/Card/Card';

const StyledTotal = styled(Card)`
  display: flex;
  justify-content: center;
`;

const Text = styled.p`
  text-transform: uppercase;
  padding: 10px 0;
  margin: 0;
`;

const TotalText = styled.span`
  font-weight: 700;
`;

const Total = ({ total }) => (
  <StyledTotal>
    <Text>
      Total Sum:&nbsp;
      <TotalText>{ total }</TotalText>
    </Text>
  </StyledTotal>
);

Total.propTypes = {
  total: PropTypes.number.isRequired,
};

export default Total;
