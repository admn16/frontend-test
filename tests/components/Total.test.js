import React from 'react';
import renderer from 'react-test-renderer';
import Total from 'components/Total/Total';

describe('<Total/>', () => {
  const props = {
    total: 10,
  };

  test('renders correctly', () => {
    const tree = renderer
      .create(<Total {...props} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
