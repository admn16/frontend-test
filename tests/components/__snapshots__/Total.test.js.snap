// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Total/> renders correctly 1`] = `
<div
  className="Total__StyledTotal-i597gn-0 glAgzs Card__StyledCard-hu4gdg-0 hNyZiT"
>
  <p
    className="Total__Text-i597gn-1 gYVLGj"
  >
    Total Sum:Â 
    <span
      className="Total__TotalText-i597gn-2 hhdlqZ"
    >
      10
    </span>
  </p>
</div>
`;
